src/Darwin/KernelNotification.cs

```diff
@@ -175,11 +179,62 @@ protected virtual void Dispose (bool disposing)
 		}
 
 		[DllImport (Constants.SystemLibrary)]
-		unsafe extern static int /* int */ kevent (KernelEvent *changeList, int /* int */ nChanges, KernelEvent *eventList, int /* int */ nEvents, IntPtr timeout);
+		unsafe extern static int /* int */ kevent (int kq, KernelEvent *changeList, int /* int */ nChanges, KernelEvent *eventList, int /* int */ nEvents, IntPtr timeout);
 		
 		[DllImport (Constants.SystemLibrary)]
-		unsafe extern static int /* int */ kevent (KernelEvent *changeList, int /* int */ nChanges, KernelEvent *eventList, int /* int */ nEvents, ref TimeSpec timeout);
+		unsafe extern static int /* int */ kevent (int kq, KernelEvent *changeList, int /* int */ nChanges, KernelEvent *eventList, int /* int */ nEvents, ref TimeSpec timeout);
+
+		public int KEvent (KernelEvent[] changeList, KernelEvent[] eventList, TimeSpan? timeout = null)
+		{
+			if (changeList == null)
+				throw new ArgumentNullException (nameof (changeList));
+
+			if (eventList == null)
+				throw new ArgumentNullException (nameof (eventList));
+
+			return KEvent (changeList, changeList.Length, eventList, eventList.Length, ToTimespec (timeout));
+		}
 
+		public unsafe int KEvent (KernelEvent[] changeList, int nChanges, KernelEvent[] eventList, int nEvents, TimeSpec? timeout = null)
+		{
+			if (changeList == null)
+				throw new ArgumentNullException (nameof (changeList));
+
+			if (eventList == null)
+				throw new ArgumentNullException (nameof (eventList));
+
+			if (changeList.Length < nChanges)
+				throw new ArgumentOutOfRangeException ("nChanges is larger than the number of elements in changeList", nameof (nChanges));
+
+			if (eventList.Length < nEvents)
+				throw new ArgumentOutOfRangeException ("nChanges is larger than the number of elements in changeList", nameof (nEvents));
+
+			unsafe {
+				fixed (KernelEvent *cp = &changeList [0])
```

Comment:
incorrect, there might be 0 items in the array and the address of `[0]` won't be valid.